mac/setup/textinputsource/main.c

```diff
@@ -0,0 +1,144 @@
+/*    
+    file
+        main.c
+
+    function
+        to manipulate text input source, i.e., 
+            - print currently selected source,
+            - select specified source (enable it as needed)
+            - enable specified source, 
+            - disable specified source,
+            - toggle enabled/disabled state of specified source.
+
+    compile
+        gcc -framework Carbon -o textinputsource main.c
+
+    usage e.g.
+        ./textinputsource [-s|e|d|t name]
+
+        given no arguments, it will print the current source name.
+        options:
+            -s : select source name (enable it as needed)
+            -e : enable source name
+            -d : disable source name
+            -t : toggle enabled/disabled on source name
+
+
+    Originally found at https://discussions.apple.com/thread/5610262
+*/
+
+#include <Carbon/Carbon.h>
+#include <libgen.h>    // basename
+
+TISInputSourceRef getInputSourceByName(char *);
+
+int
+main (int argc, char * argv[])
+{
+    int ret = -1;
+    int c;
+    TISInputSourceRef tis;
+    CFStringRef name;
+    OSStatus err = 0;
+
+    while ((c = getopt(argc, argv, "s:e:d:t:h")) != -1)
```

Comment:
Do you count `?` as a valid parameter? I see it treated as `h` at l. 92

Reply:
Yeah, I didn't write this code so haven't gone through line-by-line for this ☹ I probably should, I guess. For now, I'm okay with this being somewhat incomplete -- the desire is to rewrite the whole installer in a future version, with this being a stopgap.